omod/src/main/webapp/resources/scripts/conditionlist/controllers/addcondition.controller.js

```diff
@@ -31,25 +35,46 @@ function ConditionController($scope, RestfulService, ConditionModel, ConceptMode
                 emr.errorAlert("Error Saving condition");
             });
         }
+    }
 
     self.initCondition = self.initCondition || function () {
             $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
             $scope.condition = new ConditionModel($scope.patientUuid);
+            self.showEndDate();
         }
 
-    self.validateCondition = self.validateCondition || function () {
-            var concept = $scope.concept.concept;
-            if (concept !== null) {
-                $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
-            } else {
-                // noncoded diagnosis
-                $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
-            }
-
-            $scope.condition.onSetDate = self.getSelectedDate();
+    self.showEndDate = self.showEndDate || function() {
+        var groups = document.getElementsByClassName("group");
+        if($scope.condition.status=='INACTIVE'){
+            groups[2].style.visibility = "visible";
+        }
+        else{
+            groups[2].style.visibility = "hidden";    
+        }
+     }   
 
-            self.saveCondition();
+    self.validateCondition = self.validateCondition || function () {
+        var concept ;
+        if($scope.concept.concept.uuid)
```

Comment:
Have all these new changes got anything to do with the ticket?

Reply:
It is to validate the information before the condition gets saved.

Reply:
As we also have to validate the end date for inactive conditions, I had to make those changes for validation of end date.